scripts/lib.sh

```diff
@@ -316,3 +379,142 @@ detect_go_version()
 	[ "$?" == "0" ] && [ "$GO_VERSION" != "null" ]
 }
 
+detect_rust_version()
+{
+	info "Detecting agent rust version"
+	typeset -r yq=$(command -v yq || command -v ${GOPATH}/bin/yq)
+	if [ -z "$yq" ]; then
+		source "$yq_file"
+	fi
+
+	local runtimeRevision=""
+
+	# Detect runtime revision by fetching the agent's VERSION file
+	local runtime_version_url="https://raw.githubusercontent.com/kata-containers/agent/${AGENT_VERSION:-master}/VERSION"
+	info "Detecting runtime version using ${runtime_version_url}"
+
+	if runtimeRevision="$(curl -fsSL ${runtime_version_url})"; then
+		[ -n "${runtimeRevision}" ] || die "failed to get agent version"
+		typeset -r runtimeVersionsURL="https://raw.githubusercontent.com/kata-containers/runtime/${runtimeRevision}/versions.yaml"
+		info "Getting rust version from ${runtimeVersionsURL}"
+		# This may fail if we are a kata bump.
+		if RUST_VERSION="$(curl -fsSL "$runtimeVersionsURL" | tac | tac | $yq r - "languages.rust.version")"; then
```

Comment:
Isn't `tac | tac` the same as `cat` (or a nop in this case)? Confused why we need this.

Reply:
I don't know. I copied it from somewhere.. :)

Reply:
fwics you can remove both those `tacs` as "two tacs" == "NOP":

```sh
$ foo=$(echo -e "line 1\nline 2\nline 3")
$ echo $foo
line 1
line 2
line 3
$ echo "$foo"|tac
line 3
line 2
line 1
$ echo "$foo"|tac|tac
line 1
line 2
line 3
```

Same comment for the other `tac`s below.